whisper/src/bin/whisper-fetch.rs

```diff
@@ -87,33 +84,31 @@ fn main() -> Result<(), Error> {
         Some(ref s) => return Err(format_err!("No such drop option {}.", s)),
     };
 
-    let values: Vec<String> = archive
-        .values
-        .into_iter()
-        .filter(&filter)
-        .map(|v| v.map(|x| x.to_string()).unwrap_or("None".to_owned()))
-        .collect();
+    let archive = file.fetch(seconds_per_point, interval, now)?
+        .ok_or(err_msg("No data in selected timerange"))?
+        .filter_out(&filter);
 
     if args.json {
-        let john = json!({
-            "start": archive.from_interval,
-            "end": archive.until_interval,
-            "step": archive.step,
-            "values": values,
-            });
-
-        println!("{}", serde_json::to_string_pretty(&john)?);
+        println!("{}", serde_json::to_string_pretty(&archive)?);
     } else {
-        for (index, value) in values.iter().enumerate() {
+        for (index, value) in archive.values.iter().enumerate() {
```

Comment:
This loop looks sub-optimal.

Reply:
I am not sure I understand you :)
Any details or alternatives?